data_pipelines_cli/cli_commands/publish.py

```diff
@@ -0,0 +1,124 @@
+import glob
+import pathlib
+from typing import Any, Dict, List, Tuple
+
+import click
+import yaml
+
+from ..cli_constants import BUILD_DIR, PROFILE_NAME_ENV_EXECUTION
+from ..cli_utils import echo_subinfo
+from ..config_generation import get_profiles_yml_build_path
+from ..data_structures import DbtModel, DbtSource
+from ..dbt_utils import read_dbt_vars_from_configs
+from ..errors import DataPipelinesError
+from .prepare_env import replace_profiles_vars_with_values
+
+DBT_DATABASE_KEY_BY_PROVIDER = {
+    "bigquery": "project",
+    "postgres": "dbname",
+    "redshift": "dbname",
+    "snowflake": "database",
+}
+
+
+def _get_project_name_and_version() -> Tuple[str, str]:
+    with open(pathlib.Path.cwd().joinpath("dbt_project.yml"), "r") as f:
+        dbt_project_config = yaml.safe_load(f)
+        return dbt_project_config["name"], dbt_project_config["version"]
+
+
+def _get_database_name(env: str) -> str:
+    with open(
+        get_profiles_yml_build_path(PROFILE_NAME_ENV_EXECUTION), "r"
+    ) as profiles_yml:
+        profiles_dict = replace_profiles_vars_with_values(
+            yaml.safe_load(profiles_yml), read_dbt_vars_from_configs(env)
+        )
+
+    if len(profiles_dict) > 1:
+        raise DataPipelinesError(
+            "Generated 'profiles.yml' file has more than one profile listed."
+        )
+
+    profile_key = next(iter(profiles_dict))
+    profile = profiles_dict[profile_key]
+    profile_output = profile["outputs"][profile["target"]]
+    return profile_output[DBT_DATABASE_KEY_BY_PROVIDER[profile_output["type"]]]
+
+
+def _parse_models_schema() -> List[DbtModel]:
+    with open(pathlib.Path.cwd().joinpath("dbt_project.yml"), "r") as f:
+        dbt_project_config = yaml.safe_load(f)
+        # TODO: in dbt 1.0 it is model-paths
+        source_paths = dbt_project_config["source-paths"]
+
+    models = []
+
+    for source_path in source_paths:
+        model_path = pathlib.Path.cwd().joinpath(source_path)
+
+        # According to the dbt docs, 'schema.yml' does not have to be named 'schema'
+        yml_files = glob.glob(str(model_path.joinpath("*.yml")))
+
+        if len(yml_files) == 0:
+            raise DataPipelinesError(f"No .yml file in {model_path}.")
+        for schema_yml_name in yml_files:
+            with open(schema_yml_name, "r") as schema_yml:
+                echo_subinfo(f"Processing {schema_yml_name}.")
+                models.extend(yaml.safe_load(schema_yml).get("models", iter(())))
```

Comment:
I'm thinking what happen if model will be created as sql file but no description in yml will be created. But maybe we can have this limitation and force description as a good practice.